tensorflow/contrib/cmake/tf_shared_lib.cmake

```diff
@@ -94,3 +94,42 @@ endif()
 if(WIN32)
   add_dependencies(tensorflow tensorflow_static)
 endif(WIN32)
+
+install(TARGETS tensorflow
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
+
+# install headers
+get_property(include_dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+# remove root directories, since they contain everything
+list(REMOVE_DUPLICATES include_dirs)
+list(REMOVE_ITEM include_dirs ${tensorflow_source_dir} ${CMAKE_CURRENT_BINARY_DIR})
+# append core tensorflow header dirs and binary dir where generated headers reside
```

Comment:
I think this is a superset of the public headers—or at least, I can't see where the filtering happens. Can you try to restrict it to the same set of headers as is installed with the PIP package? (AIUI, you'll also need to install some of the protobuf public headers, if they aren't already installed.)

Reply:
Thanks @mrry for the quick review. Since I was not quite sure which headers to install and it probably depends on the CMake options I've created universal solution which just installs all the headers used by tensorflow package with same directory structure. This way you also don't need to worry about adjusting this code when tensorflow package is changed. If you confirm that it is unlikely that more packages will be added to public headers set, then I'll try to restrict amount of headers installed. I know for sure that Eigen, Protobuf, nsync and protobuf generated headers are needed to compile trivial C++ program using tensorflow.

Reply:
It's not unlikely that more headers will be added to the set of public headers in future changes, but the PIP package generated by the Bazel build is a reliable source of truth for what headers need to be included. (IIRC @av8ramit was looking into including these headers in the CMake build at one point, so perhaps he has something you could reuse?)

Reply:
https://github.com/tensorflow/tensorflow/commit/fd584b5e9daf57c827b0ee652317ab0f205ae4bb#diff-10f12827f7259a4fd7dba7abbd8307b0 Hopefully this helps?

Reply:
Oh sweet, I didn't know that was already in! (I think it might need to be updated for nsync, but apart from that it's just what we need.)

Reply:
Thanks @av8ramit it was indeed helpful! @mrry I've added install targets based on the @av8ramit solution, needed to add some extra headers from tensorflow/cc and nsync.